components/backup/src/writer.rs

```diff
@@ -0,0 +1,207 @@
+// Copyright 2019 TiKV Project Authors. Licensed under Apache-2.0.
+
+use std::sync::Arc;
+use std::time::Instant;
+
+use engine::rocks::{SstWriter, SstWriterBuilder};
+use engine::{CF_DEFAULT, CF_WRITE, DB};
+use external_storage::ExternalStorage;
+use kvproto::backup::File;
+use tikv::raftstore::store::keys;
+use tikv::storage::txn::TxnEntry;
+use tikv_util;
+
+use crate::metrics::*;
+use crate::{Error, Result};
+
+/// A writer writes txn entries into SST files.
+pub struct BackupWriter {
+    name: String,
+    default: SstWriter,
+    default_written: bool,
+    write: SstWriter,
+    write_written: bool,
+}
+
+impl BackupWriter {
+    /// Create a new BackupWriter.
+    pub fn new(db: Arc<DB>, name: &str) -> Result<BackupWriter> {
+        let default = SstWriterBuilder::new()
+            .set_in_memory(true)
+            .set_cf(CF_DEFAULT)
+            .set_db(db.clone())
+            .build(name)?;
+        let write = SstWriterBuilder::new()
+            .set_in_memory(true)
+            .set_cf(CF_WRITE)
+            .set_db(db.clone())
+            .build(name)?;
+        let name = name.to_owned();
+        Ok(BackupWriter {
+            name,
+            default,
+            default_written: false,
+            write,
+            write_written: false,
+        })
+    }
+
+    /// Wrtie entries to buffered SST files.
+    pub fn write<I>(&mut self, entries: I) -> Result<()>
+    where
+        I: Iterator<Item = TxnEntry>,
+    {
+        for e in entries {
+            match e {
+                TxnEntry::Commit { default, write } => {
+                    // Default may be empty if value is small.
```

Comment:
Right, they are filtered in TxnEntryScanner.

Reply:
Should Delete be filtered out? How is these key deleted on the backup side?
Btw please also note that if in the future you backups locks(not committed data), you should keep the `Lock` type of writes, because they can be the primary key of transactions. Rest LGTM.

Reply:
For now, delete is filtered out, because we only support full backup.

We will never backup Locks.